src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/core/abstract_actor.cc

```diff
@@ -0,0 +1,94 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+// See: https://spdx.org/licenses/
+
+#include <message_infrastructure/csrc/core/abstract_actor.h>
+#include <message_infrastructure/csrc/core/utils.h>
+#include <message_infrastructure/csrc/core/message_infrastructure_logging.h>
+
+namespace message_infrastructure {
+
+AbstractActor::AbstractActor(AbstractActor::TargetFn target_fn)
+  : target_fn_(target_fn) {
+  this->ctl_shm_ = GetSharedMemManagerSingleton()
```

Comment:
Remove "this->" here and elsewhere, since it's redundant.

The underscore at the end of the member "ctl_shm_" usually implies it is a member variable already.